applications/FluidDynamicsApplication/custom_utilities/cfd_utilities.cpp

```diff
@@ -0,0 +1,425 @@
+    KRATOS_ERROR_IF(!rModelPart.HasNodalSolutionStepVariable(VELOCITY))
+        << rModelPart.Name() << " doesn't have " << VELOCITY.Name()
+        << " in nodal solution step variable list. Please add it first.\n";
+    KRATOS_ERROR_IF(!rModelPart.HasNodalSolutionStepVariable(MESH_VELOCITY))
+        << rModelPart.Name() << " doesn't have " << MESH_VELOCITY.Name()
+        << " in nodal solution step variable list. Please add it first.\n";
+
+    const std::function<double(
+        array_1d<double, 3>&, const GeometryType&, const array_1d<double, 3>&,
+        const Vector&, const double, const double, const double)>
+        method = [VonKarman, WallSmoothness, MaxIterations, Tolerance](
+                     array_1d<double, 3>& rFrictionVelocity,
+                     const GeometryType& rGeometry, const array_1d<double, 3>& rNormal,
+                     const Vector& rGaussShapeFunctions, const double Density,
+                     const double KinematicViscosity, const double WallHeight) {
+            array_1d<double, 3> r_wall_velocity;
+            GeometryUtils::EvaluateHistoricalVariableValueAtGaussPoint(
+                r_wall_velocity, rGeometry, VELOCITY, rGaussShapeFunctions);
+            array_1d<double, 3> r_mesh_velocity;
+            GeometryUtils::EvaluateHistoricalVariableValueAtGaussPoint(
+                r_mesh_velocity, rGeometry, MESH_VELOCITY, rGaussShapeFunctions);
+
+            return CalculateLinearLogarithmicWallFunctionBasedYPlusAndUtau(
+                rFrictionVelocity, r_wall_velocity - r_mesh_velocity, rNormal, KinematicViscosity,
+                WallHeight, VonKarman, WallSmoothness, MaxIterations, Tolerance);
+        };
+
+    CalculateYPlusAndUTauForConditions(rModelPart, rKinematicViscosityVariable, method);
+
+    KRATOS_CATCH("");
+}
+
+template <typename TDataType>
+void DistributeConditionVariableToNodes(ModelPart& rModelPart, const Variable<TDataType>& rVariable)
+{
+    KRATOS_TRY
+
+    CalculateNumberOfNeighbourConditions(rModelPart);
+
+    ModelPart::NodesContainerType& r_nodes = rModelPart.Nodes();
+    VariableUtils().SetNonHistoricalVariableToZero(rVariable, r_nodes);
+
+    block_for_each(rModelPart.Conditions(), [rVariable](ConditionType& r_condition) {
+        GeometryType& r_geometry = r_condition.GetGeometry();
+
+        const TDataType& r_value = r_condition.GetValue(rVariable);
+        for (unsigned int i_node = 0; i_node < r_geometry.PointsNumber(); ++i_node)
+        {
+            NodeType& r_node = r_geometry[i_node];
+            const TDataType& r_current_value = r_node.GetValue(rVariable);
+            const double number_of_neighbour_conditions =
+                static_cast<double>(r_node.GetValue(NUMBER_OF_NEIGHBOUR_CONDITIONS));
+
+            r_node.SetLock();
+            r_node.SetValue(rVariable, r_current_value + r_value * (1.0 / number_of_neighbour_conditions));
+            r_node.UnSetLock();
+        }
+    });
```

Comment:
I'd add this to the variable utils in the core as it could be useful for non-CFD purposes too.

Reply:
done in #7146 :)